### PR TITLE
Convert to cargo workspace and integrate wolfssl-sys

### PR DESCRIPTION
**Pull request by xv-ian-c**
Some of the adjustments were taken from https://github.com/expressvpn/wolfssl-sys/pull/53.

### CODE REVIEW

File: wolfssl-sys/Cargo.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ authors = ["pete.m@expressvpn.com"]
 license = "GPL-2.0"
 readme = "README.md"
 description = "System bindings for WolfSSL"
-repository = "https://github.com/expressvpn/wolfssl-sys"
+repository = "https://github.com/expressvpn/wolfssl"
```
Comment (kp-mariappan-ramasamy) [inline wolfssl-sys/Cargo.toml]: wolfssl -> wolfssl-rs

**Review by kp-mariappan-ramasamy — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,11 +1,22 @@
...
 
-This repository attempts to build safe and idiomatic abstractions for the
-[WolfSSL Embedded SSL/TLS Library (C)][wolfssl-home]. It does so by building on
-top of autogenerated bindings created by [`wolfssl-sys`][wolfssl-sys].
+This repository attempts to build safe and idiomatic abstractions for the [WolfSSL Embedded SSL/TLS Library (C)][wolfssl-home].
+
+There are two parts to this:
+
+- The [`wolfssl-sys`][] crate auto-generates unsafe Rust bindings through [bindgen], to C functions provided by the WolfSSL library.
+- The [`wolfssl`][] create then build safe and idiomatic abstractions on top of the unsafe layer.
```
[the remark this answers is not in the document]
Comment (xv-ian-c) [reply README.md:8]: You wouldn't believe how often I make that typo...

File: wolfssl-sys/Cargo.toml (hunk 1 above)
Comment (xv-ian-c) [reply wolfssl-sys/Cargo.toml]: This is handled later in the branch.

**Review by kp-mariappan-ramasamy — APPROVED** [review]
LGTM

**Review by xv-pete-m — APPROVED** [review]

**Pull request merged**